### PR TITLE
task: update dependency for php8

### PR DESCRIPTION
**Pull request by hhoechtl**
`thenetworg/oauth2-azure` is not compatible with PHP8

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
         }
     ],
     "require": {
-        "php": ">=7.4",
+        "php": ">=7.4 | 8.*",
```
Comment (jkrzefski) [inline composer.json:35]: This seems superfluous.

File: composer.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
         "psr/http-client": "^1.0",
         "psr/http-message": "^1.0",
         "psr/log": "^1.1",
-        "shopware/core": "^6.4.11",
+        "shopware/core": "~v6.4.0",
```
Comment (jkrzefski) [inline composer.json:49]: Afaik this is on purpose. Shopware broke some backwards compatibility in their administration javascript with this version, so we only support versions from this point on. But I'll have to check with @JoshuaBehrens to make sure, I remember this correctly.

File: composer.json [hunk 3]
```diff
@@ -55,14 +55,14 @@
         "symfony/options-resolver": ">=5.4 <6",
         "symfony/psr-http-message-bridge": "^2",
         "symfony/routing": ">=5.4 <6",
-        "thenetworg/oauth2-azure": "^1.4",
+        "thenetworg/oauth2-azure": "^2.0",
```
Comment (jkrzefski) [inline composer.json:58]: I understand this is the main reason for your contribution. As this is a major upgrade, we will have to test it thoroughly to make sure, everything still works. So please give us some time here 😉

File: composer.json [hunk 4]
```diff
@@ -55,14 +55,14 @@
...
         "web-token/jwt-core": "^2.2",
         "web-token/jwt-signature": "^2.2",
         "web-token/jwt-signature-algorithm-ecdsa": "^2.2",
         "web-token/jwt-signature-algorithm-rsa": "^2.2"
     },
     "require-dev": {
-        "shopware/administration": "^6.4.11"
+        "shopware/administration": "~v6.4.0"
```
Comment (jkrzefski) [inline composer.json:65]: See my other comment. This is on purpose.

File: src/Component/Provider/DefaultInjectingEntityRepositoryDecorator.php [hunk 5]
```diff
@@ -63,9 +63,9 @@ public function create(array $data, Context $context): EntityWrittenContainerEve
         return $this->decorated->create($this->injectDefaults($data, $context), $context);
     }
 
-    public function delete(array $data, Context $context): EntityWrittenContainerEvent
+    public function delete(array $ids, Context $context): EntityWrittenContainerEvent
```
Comment (jkrzefski) [inline src/Component/Provider/DefaultInjectingEntityRepositoryDecorator.php:66]: It makes sense to adopt the argument names of the interface. I can imagine, this improves support for [named arguments](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments).

I'll leave this link here for reference: https://github.com/shopware/platform/blob/v6.4.11.1/src/Core/Framework/DataAbstractionLayer/EntityRepositoryInterface.php#L31

**Review by jkrzefski — COMMENTED** [review]
Thank you for your contribution. We will look into this in the next days and hopefully we can merge it soon.

**Comment by JoshuaBehrens** [comment]
I don't want to question the need of this change. I see that the package is not compatible but we use it on environments where we run php 8. How do you use it and how did you find that issue? This would be crucial to know for our testing.

**Comment by hhoechtl** [comment]
We are very strict with our composer configuration regarding PHP versions:

```json
{
    "require": {
        "php": "^8.0"
    },
    "config": {
        "platform": {
            "php": "8.0.10"
        }
    },
```

So we're sure the package maintainers know what they are doing. 
With this configuration we can't install your OAuth Bundle as the package `thenetworg/oauth2-azure` in version 1.x is only flagged compatible with PHP7.

```json
    "require": {
        "php": "^5.6|^7.0",
```

**Review by silviokennecke — COMMENTED** [review]
For me okay. Could you please update the Changelog?

**Comment by JoshuaBehrens** [comment]
@hhoechtl for now it looks good to us. We suggest to allow installation of both packages as well php 8.0. We will further have to provide the ZIP file in the community store as php7.4. Does this work out for you?

**Comment by hhoechtl** [comment]
Sure

**Review by JoshuaBehrens — APPROVED** [review]

**Comment by JoshuaBehrens** [comment]
I also added changelog entries and some CI tests for the php8.0 and php8.1 installations :)

**Pull request merged**